src/Microsoft.AspNetCore.Routing/EndpointRoutingMiddleware.cs

```diff
@@ -77,6 +77,20 @@ public async Task Invoke(HttpContext httpContext)
             await _next(httpContext);
         }
 
+        private static void SetRouteValuesFeature(HttpContext httpContext)
```

Comment:
Disclaimer: I'm not sure yet whether this would even work in The Real World. Need to add a functional test for accessing and setting route values earlier in the pipeline before `UseEndpointRouting` is hit.